uniffi_bindgen/src/interface/function.rs

```diff
@@ -90,6 +90,21 @@ impl Function {
         self.ffi_func.return_type = self.return_type.as_ref().map(|rt| rt.into());
         Ok(())
     }
+
+    // Intentionally exactly the same as the Method version
+    pub fn type_contains_unsigned_types(&self, ci: &ComponentInterface) -> bool {
+        let check_return_type = {
+            match self.return_type() {
+                None => false,
+                Some(t) => ci.type_contains_unsigned_types(t),
+            }
+        };
```

Comment:
A fun Rust thing you could try here, is to turn `return_type()` into an iterator and chain it with the arguments to check them all at once. Something like:

```
self.return_type().iter()
    .chain(self.arguments().iter().map(|arg|, arg.type_))
    .any(|t| ci.type_contains_unsigned_type(t))
```

It might turn out to be a bit awkward with the types because `return_type()` returns a borrow but `arg.type_()` returns a clone, but it's be interesting to try.

(To be clear: the existing code you have here is also fine to merge as-is, this is purely a "hey, good opportunity to learn more about Rust iterators if you're interested!" situation).

Reply:
oh interesting! I couldn't seem to get it to properly compile after time-boxing however feel pretty good about using chain in the future with how much I tried to wrangle with it lol